dbms/src/test/java/org/polypheny/db/mql/AggregateTest.java

```diff
@@ -175,222 +176,279 @@ public void groupNullTest() {
 
         insertMany( DATA_2 );
 
-        Result result = aggregate( $replaceRoot( "$test.key" ) );
+        DocResult result = aggregate( $replaceRoot( "$test" ) );
+
+        MongoConnection.checkDocResultSet( result, expected, false, true );
+    }
+
 
-        MongoConnection.checkUnorderedResultSet( result, expected, false );
+    @Test
+    public void replaceRootNotDocTest() {
+        try {
+            insertMany( DATA_2 );
+
+            DocResult result = aggregate( $replaceRoot( "$test.key" ) );
+            Assertions.fail();
+        } catch ( Exception e ) {
+            // empty on purpose
+        }
     }
 
     //$replaceWith
 
 
     @Test
     public void replaceWithTest() {
-        List<String[]> expected = ImmutableList.of(
-                new String[]{ "1" },
-                new String[]{ null },
-                new String[]{ "13" } );
+        List<String> expected = Arrays.asList(
+                document( kv( string( "key" ), 1 ) ),
+                document( kv( string( "key1" ), 5 ) ),
+                document( kv( string( "key" ), 13 ) ) );
+
+        insertMany( DATA_2 );
+
+        DocResult result = aggregate( $replaceWith( "$test" ) );
+
+        MongoConnection.checkDocResultSet( result, expected, false, true );
+    }
+
+
+    @Test
+    public void replaceWithNonDocTest() {
 
         insertMany( DATA_2 );
 
-        Result result = aggregate( $replaceWith( "$test.key" ) );
+        try {
+            DocResult result = aggregate( $replaceWith( "$test.key" ) );
 
-        MongoConnection.checkUnorderedResultSet( result, expected, false );
+            // this has to fail
+            Assertions.fail();
+        } catch ( Exception e ) {
+            // empty on purpose
+        }
     }
 
     //$set
 
 
     @Test
     public void setTest() {
-        List<String[]> expected = ImmutableList.of(
-                new String[]{ "{\"test\":1,\"testing\":\"entry\"}" },
-                new String[]{ "{\"test\":1.3,\"key\":{\"key\":\"val\"},\"testing\":\"entry\"}" },
-                new String[]{ "{\"test\":\"test\",\"key\":13,\"testing\":\"entry\"}" } );
+        List<String> expected = ImmutableList.of(
+                "{\"test\":1,\"testing\":\"entry\"}",
+                "{\"test\":1.3,\"key\":{\"key\":\"val\"},\"testing\":\"entry\"}",
+                "{\"test\":\"test\",\"key\":13,\"testing\":\"entry\"}" );
 
         insertMany( DATA_0 );
 
-        Result result = aggregate( $set( "{\"testing\": \"entry\"}" ) );
+        DocResult result = aggregate( $set( "{\"testing\": \"entry\"}" ) );
 
-        MongoConnection.checkUnorderedResultSet( result, expected, true );
+        MongoConnection.checkDocResultSet( result, expected, true, true );
     }
 
     //$skip
 
 
     @Test
-    @Category(CottontailExcluded.class) // cottontail does not support skips/offset queries
+    @Tag("cottontailExcluded") // cottontail does not support skips/offset queries
```

Comment:
Why is this exclude necessary? Query should be processed in Polypheny.

Reply:
Artifact, removed.